src/node/rpc/node_frontend.h

```diff
@@ -1377,34 +1386,41 @@ namespace ccf
             "Configurations: sufficient number of ORCs, updating nodes in "
```

Comment:
I'm a little worried about the call to `consensus->orc()` on line 1383. This endpoint writes to the KV and can thus be aborted or rolled back. However, the endpoint also has some side effects on the internal state of the consensus (`orc_sets`) which does not seem to be transactional in nature (e.g. I do not see any rollback logic for it). 

I think `orc_sets` should be recorded in the KV store instead (e.g. whether it's passed the quorum or not could be recorded as a `bool` in the `configuration` table), which then passed to the consensus via a hook (which might not be necessary as we already have a hook on the configuration table).

Reply:
As discussed, this should be OK as long as i) nodes retry to send ORCs until they've seen their `TRUSTED`ness committed, ii) `orc_sets` (and other consensus state updated in `consensus->orc()` are flushed when a node becomes candidate. We should also try to add a test for this (end-to-end or unit test?) but it's OK to merge this PR as it is as long as the existing reconfiguration test is stable.

Reply:
I created #3157 for this and assigned myself, so I'll resolve the conversation here.